services/catarse/spec/decorators/contribution_detail_decorator_spec.rb

```diff
@@ -91,4 +91,16 @@
       it { is_expected.to eq('http://foo.bar/') }
     end
   end
+
+  describe '#display_pix_qr_code' do
+    let(:contribution) { create(:confirmed_contribution) }
+    context 'when pix_qr_code is filled' do
+      before do
+        payment.update gateway_data: { pix_qr_code: 'http://foo.bar/' }
+      end
+
+      subject { detail.decorate.display_pix_qr_code }
+      it { is_expected.to be_a_kind_of(String) }
```

Comment:
**Sugestão**
Aqui tu pode fazer: `it { is_expected.to eq 'http://foo.bar/' }`. É mais seguro que testar só o tipo da resposta, pois se alguém modificar o `display_pix_qr_code` pra retornar uma string aleatória ou outro campo, esse teu teste vai continuar passando.

Reply:
@stephannv aqui no caso é gerado um XML que é o qr_code. Como eu procederia nesse caso?

Reply:
@gabrielras , nesse caso, tanto faz pois no `before` tu está fazendo um `payment.update gateway_data: { pix_qr_code: 'http://foo.bar/' }`, ou seja, o `payment.gateway_data['pix_qr_code']` vai retornar o que tu colocou nesse update. Se quiser mudar pra uma string aleatória pode mudar, o lance é que teu expect tem que verificar se ele tá retornando a string que tu colocou, e não uma string qualquer.